src/main/java/ch/uzh/ifi/hase/soprafs21/controller/UsersApiController.java

```diff
@@ -175,6 +171,20 @@ public UsersApiController(NativeWebRequest request, UserService userService, Cha
                 .body(userLoginPostDto);
     }
 
-
-
+    @Override
+    public ResponseEntity<UserDto> usersUserIdGet(Long userId) throws Exception {
+        UserDto userDto = null;
+        if (userId != null) {
+            userDto = userService.getUserDetails(userId);
+            if(null == userDto){
+                throw new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found");
+            }
+        }
+        else {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "Invalid input userId cannot be null");
```

Comment:
Unauthorized is reserved for when a user is not allowed to take some action, 400 BAD REQUEST would be more appropriate here

Reply:
updated